### PR TITLE
Does no show consent screen on subdomains of NSS

### LINKED ISSUE #1200: don't show consent dialog for data browser
when logging in to the databrowser, it does not make sense to show the user a consent screen. the databrowser should always have read+write+control, since it's same-origin.

### PR DESCRIPTION
**Pull request by jaxoncreed**
Fix #1200

### CODE REVIEW

File: lib/requests/consent-request.js [hunk 1]
```diff
@@ -66,11 +66,13 @@ class ConsentRequest extends AuthRequest {
...
 
-    const appOrigin = request.getAppOrigin()
+    const appUrl = request.getAppUrl()
+    const appOrigin = `${appUrl.protocol}//${appUrl.host}`
+    const serverUrl = url.parse(req.app.locals.ldp.serverUri)
     // Check if is already registered or is data browser
     if (request.isUserLoggedIn()) {
       if (
-        appOrigin === req.app.locals.ldp.serverUri ||
+        (appUrl && appUrl.host.includes(serverUrl.host) && appUrl.protocol === serverUrl.protocol) ||
```
Comment (RubenVerborgh) [inline lib/requests/consent-request.js]: server: `https://safe.com/`
app: `https://unsafe.com/`

server: `https://safe.com/`
app: `https://safe.com.evil.com/`

Comment (jaxoncreed) [reply lib/requests/consent-request.js]: Ah right

Comment (RubenVerborgh) [reply lib/requests/consent-request.js]: Tests are really important for these security cases.

Comment (jaxoncreed) [reply lib/requests/consent-request.js]: Yep, that was a lapse in judgment on my part. If we get the green light on deploying going forward with this version I'll crank out some more tests.

Though thankfully this wouldn't have been a security hole as this only controls if the consent screen pops up, and not if the app is allowed.

Comment (RubenVerborgh) [reply lib/requests/consent-request.js]: The logic should only reside in one place though.

**Comment by michielbdejong** [comment]
What's the status of this, can we get this merged before the deploy?

**Pull request merged**